DBZ-8788 Document mTLS using JKS with Oracle

Naros opened the pull request:
https://issues.redhat.com/browse/DBZ-8788

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+
+[NOTE]
+====
+It is important for TLS encryption to use `database.url` rather than `database.host`.
+The URL provides a TNS connection string which is what explicitly controls the secure connection requirement by setting the protocol to `tcps`.
```
```suggestion
For {prodname} to use TLS encryption to communicate with Oracle, requires a TCPS connection with the server.
To establish a TCPS connection, you must configure the connector to use the `database.url` property, and not the `database.host` property. 
Unlike the `database.host` property, the `database.url` property permits you to define an Oracle TNS (Transparent Network Substrate) connection string that explicitly requires use of the `TCPS` protocol. 
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
 When connecting to Oracle using mutual TLS authentication (mTLS), this involves both the connector and the database service proving their identities.
 The {prodname} for Oracle connector relies on the Oracle JDBC driver's built-in functionality to support mTLS authentication.
 
+There are two ways to establish mTLS connections:
```
```suggestion
You can use either of the following methods to establish mTLS connections between {prodname} and Oracle:
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
 
+There are two ways to establish mTLS connections:
+
+* xref:#using-mtls-secure-connections-java-key-trust-stores[]
```
```suggestion
* xref:#using-mtls-secure-connections-java-key-trust-stores[Using Java key and trust stores to configure mTLS connections].
```

roldanbob replied:
In shared content (community/product), we must provide link text, because the text is not automatically resolved if the link is used downstream.  
In this particular case, the upstream link text will be used, rather than the topic title specified by the downstream Nebel annotation, but better that than no text, and we can live with the small discrepancy as long as the link works.

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+There are two ways to establish mTLS connections:
+
+* xref:#using-mtls-secure-connections-java-key-trust-stores[]
+* xref:#using-mtls-secure-connections-oracle-wallet[]
```
```suggestion
* xref:#using-mtls-secure-connections-oracle-wallet[Using Oracle Wallet to configure mTLS connections].
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+
+* xref:#using-mtls-secure-connections-java-key-trust-stores[]
+* xref:#using-mtls-secure-connections-oracle-wallet[]
+
```
```suggestion
// Type: procedure
// ModuleID: oracle-connector-using-java-key-trust-stores-to-configure-mtls
// Title: Using Java key and trust stores to configure the Oracle connector to use mTLS
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+It is important for TLS encryption to use `database.url` rather than `database.host`.
+The URL provides a TNS connection string which is what explicitly controls the secure connection requirement by setting the protocol to `tcps`.
+====
+
```
```suggestion
// Type: procedure
// ModuleID: oracle-connector-using-oracle-wallet-to-configure-mtls
// Title: Using Oracle wallet to configure the connector to use mTLS
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+==== Java key/trust stores
+
+.Prerequisites
+* Verify connector can access key/trust store files directly.
+* Verify database TNS listener supports `tcps` secure connections.
```
```suggestion
* Verify that the connector can access the configured key and trust store files.
* Verify that the database TNS (Transparent Network Substrate) listener supports TCPS secure connections.
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+* Verify database TNS listener supports `tcps` secure connections.
+
+.Procedure
+* Configure the {prodname} Oracle connector, as illustrated below in the example.
```
```suggestion
* Configure the {prodname} Oracle connector, as illustrated in the following example. +
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+
+.Procedure
+* Configure the {prodname} Oracle connector, as illustrated below in the example.
+
```
```suggestion
+
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+.Procedure
+* Configure the {prodname} Oracle connector, as illustrated below in the example.
+
+.Example: {prodname} Oracle connector TLS configuration
```
```suggestion
.Example: {prodname} Oracle connector TLS configuration
=====
```

roldanbob commented on documentation/modules/ROOT/pages/connectors/oracle.adoc:
```diff
@@ -3299,6 +3299,44 @@ endif::community[]
+  "driver.javax.net.ssl.trustStoreType": "JKS"
+}
+----
+
```
```suggestion
=====
+
```
To use example titles, the example must be formatted as an example block.

roldanbob reviewed (changes requested):
@Naros Thanks for adding this. Reminds me that on a back burner I still have [DBZ-6168](https://issues.redhat.com/browse/DBZ-6168) for clarifying the connection string settings for XStream/TCPS. This looks to be very similar. I had written up something, but I never had a good feeling for whether it covered the required bases. Now that we're exposing the XStreams content, if you can provide verification of what needs to be done, maybe we can push that forward.

Naros commented:
Hi @roldanbob, so looking at that linked Jira, that one specifically was discussing Oracle Wallet for XStream.  The same would hold true whether the user is using LogMiner or XStream (as documented in the Wallet section).  What I've added is for mTLS using Java Keystores, which is a non-Wallet approach. This should also work for XStream, too.

We generally structured the documentation in a way to favor LogMiner but I think we might want to rethink that strategy long-term. It may pay to take one of two likely strategies; though there may be more.

* Split the documentation into General,  LogMiner, XStream, and OLR sections.  We have this mostly now except we've embedded the LogMiner content into the General content. As we want to have an emphasis on having both of these adapters as supported features, it may be wise to organize the content with equal weight.
* We tailor the content generally and then weave in each adapter's specifics into top-level generalized sections. 

I personally prefer the former, but I"d like your opinion.

roldanbob commented:
> I personally prefer the former, but I"d like your opinion.

I agree with the proposed approach. I don't think that I can address this for 3.0.8, but we can make it a priority for Debezium.next. For now, I think that you can merge this, yes? And post-3.0.8 we can work in the content for DBZ-6168 (waller or Java keystore).
